File: Practice Code/DSA_in_java/stack_implementation/ArrayStack.java
package DSA_in_java.stack_implementation;

public class ArrayStack<T> {
    static final int MAX_SIZE=30;
    T stack[]=(T[])new Object[MAX_SIZE];
    int top;
    ArrayStack(){
        top=-1;
    }

    public void push(T data){
        if(top==MAX_SIZE-1)
            throw new IndexOutOfBoundsException("Stack Overflow");
        stack[++top]=data;
    }

    public T pop(){
        if(top==-1)
            throw new IndexOutOfBoundsException("Stack Underflow");
        return stack[top--];
    }

    public T peek(){
        if(top==-1)
            throw new IndexOutOfBoundsException("Stack Underflow");
        return stack[top];
    }
    public void display(){
        if(top==-1)
            throw new IndexOutOfBoundsException("Stack Underflow");
        for(int i=top;i>=0;i--){
            System.out.println(stack[i]);
        }
    }

}

File: DSA in C/stack/ArrayStack.c
#include<stdio.h>
#define SIZE 5
int Stack[SIZE];
int top=0;

void push(int x){
    if(top==SIZE-1){
        printf("Stack is full\n");
    }
    else{
        Stack[top++]=x;
    }
}

void pop(){
    if(top==-1){
        printf("Stack is Empty\n");
    }
    else{
        printf("The poped element is %d\n",Stack[top]);
        top--;
    }
}

void display(){
    for(int i=top-1;i>=0;i--)
        printf("%d\t",Stack[i]);
}
int main()
{
    push(7);
    push(6);
    push(56);
    push(67);
    display();
    return 0;
}